lib/MusicBrainz/Server/Controller/Artist.pm

```diff
@@ -485,7 +486,7 @@ with 'MusicBrainz::Server::Controller::Role::Edit' => {
                         $c->model('EditNote')->add_note(
                             $ac_edit->id,
                             {
-                                text => "The artist name has been changed in edit #$editid.",
+                                text => "This credit is being changed because the main name for the artist qq{$artistname} is being modified by edit #$editid.",
```

Comment:
`qq{$artistname}` should just be `$artistname` (you're already inside quotes so that would literally output `qq{U2}`).

Also, should we add display quotes around `“$artistname”`? e.g. `the artist “is” is` reads a bit more clearly than `the artist is is`.

Reply:
Adding display quotes is exactly what I was trying to achieve here. Do I just use “ ” then?

Reply:
Yes, you can use the unicode ones directly in the string.

Otherwise the ASCII ones would just need to be escaped with a backslash: `artist \"$artistname\" is`.

`qq` only makes sense on the outside as an alternative quote operator, so you can do e.g. `qq("a'bunch"of'nonsense")` without having to escape those normal quotes.

Reply:
For the record, it requires both `utf8` and a backslash for a proper encoding of unicode quotes.
Fixed in https://github.com/metabrainz/musicbrainz-server/commit/9fc9a4cd99193729c720cf556937c3115f10ae1e

Reply:
Now I'm curious, what happens without the backslash?

Reply:
Fancy characters are displayed instead: https://musicbrainz.org/edit/49349519

It is only necessary for not-localized messages since encoding is done by [`l` function](https://github.com/metabrainz/musicbrainz-server/blob/master/lib/MusicBrainz/Server/Plugin/Translation.pm#L27).

Reply:
Sorry, I meant I saw the edit in IRC, but I don't understand why the backslash is needed. I never had to backslash Unicode quotes before and couldn't find what this does in any documentation, so I want to learn what it does. :)

Reply:
Sorry, I just meant it has the same weird output. I guess it is probably the same as:
`"This ... artist" . encode('utf-8', '“') . $artistname . encode('utf-8', '”') . "is ..."`

Note that this message is not passed directly to template. It is stored into the database first. That is probably why it requires special attention compared to other messages.